### PR TITLE
Fix sugar

### LINKED ISSUE #72421: Please fix sugar
### Describe the bug

Some time ago, sugar was reduced in weight from 5g to 2g, in an attempt to improve its density. But the calorie count was never reduced from 19 to 8, and other sugar-linked bits of code such as the "sugar_standard" food requirement  - "One unit [10u sugar] provides ~200kcal; chosen to match one unit of molasses" - were never made coherent. 

To be clear, this inaccuracy is a problem, as it low-key broke all recipes dependent on sugar, and it paralyzes work with any such. 

### Attach save file

Sugar

### Steps to reproduce

Sugar

### Expected behavior

So how do we fix it? 

The general information on density of sugar found on the Internet is suspicious. For one thing, there is a big mystery in how table sugar could be half as dense as sucrose, when table sugar is chemically just sucrose.

I have a sack of typical Domino sugar, a staple American brand. The dimensions are fully 170x120x100 mm, holding a net weight of 1810 g. Thus the minimum density, including packaging volume, would be 1810 g/2040 ml. That would indicate a density of 0.9 g/ml, only a little higher than most sources indicate, but still less than typically-available fresh water.

And yet, sugar will readily sink in water, as [this video](https://www.youtube.com/watch?v=ZmWYG7qh0QA) demonstrates. This  should be a widely-familiar phenomenon that demonstrates the direction of sugar's density relative to water.

Is it that bulk granulated sugar is radically less dense - half as dense - than a given molecule of it? 

Whatever the case may be, it is up to the community to fix sugar. 

### Screenshots

_No response_

### Versions and configuration

Sugar

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Montimorency**
#### Summary
Balance "Fix sugar"

#### Purpose of change
Closes #72421
Revises the sugar weight change from #70176 

Someone had to act. Let the wars begin. 

#### Describe the solution
Sugar can't be as energy-dense as fat anymore. 

Since time immemorial, C:DDA has known sugar to be of 5 "weight." Balance must be restored.

5/6 is a compromise ratio, rounded, that restores the traditional sugar component in most crafting recipes and realigns sugar weight to what is represented in the sugar_standard Cooking Requirement

```
"id": "sugar_standard",
    "type": "requirement",
    "//": "One unit provides ~200kcal; chosen to match one unit of molasses",
    "components": [
      [
        [ "sugar", 10 ],
```

while staying within the bounds of commonly-cited densities - 0.8 +/- 0.1 -  without litigating them. This will also restore the discrepancy of about a calorie in the unitary sugar due to strict rounding (19 rather than 20), for all the sugar skeptics out there. 

#### Describe alternatives you've considered
Reverting #70716 is too brute force, and sugar's density _was_ wrong. 

My preferred relative density would be a clean one (1), at 5:5. 

Equating a unit of sugar to a cooking teaspoon, i.e. 4:5 is another compromise, but it still demands the work of adjusting calories, adjusting all recipes that rely on sugar, and reworking the sugar_standard food requirement for new quantity. It also ascribes a density range that's a bit too low for my sensibility. 

#### Testing
Loaded saves. The overflow error codes with existing containers on saves have been nixed. 

Debug tested sugar jar item groups. The maximum spawn in a 0.5 L jar is now 83 sugar.

#### Additional context
Two item groups, sugar_jar_glass_sealed_10_142 and sugar_jar_glass_sealed_10_120, had obnoxious (though dismissable) error codes on existing saves suppressed by addition of overflow conditionals. 

MRE package was slightly increased in carrying volume to accommodate larger sugar packets.

### CODE REVIEW

**Comment by Montimorency** [comment]
Comment: 
@wwkk222208 

You increased MRE packaging size a month ago in #71529, and added higher contained spawns for sugar (up to 10 units). This was done, and probably tested, under the new volume of sugar introduced in December. I have a stash of MREs on one of my saves, and the maximum-sized sealed ones I've got spawn at 1.59 L of contents. Raising the volume of sugar causes an overflow error with the heat pack. Item group debug doesn't allow me to peek at interiors, so I can't tell how restoring sugar volume would interact with MRE spawns going forward. It seems possible that future MRE spawns will be dropping heat packs unintendedly, rather than lowballing sugar. Could you test that?

**Comment by wwkk222208** [comment]
I didn't actually find the weight of sugar in MRE, and the 10 portions of sugar in this PR were determined based on the sugar water recipe, because it doesn't make sense to have unusable materials in meal ready to eat. However, I now think there shouldn't be so much sugar in MRE, and a better solution would be to reduce the sugar content and add a recipe for "lightly sweetened water".
Regarding the packaging of MRE: The capacity of the outer packaging needs to be exactly the same as the contents. If it is larger than the contents, the MRE cannot be sealed; if it is smaller, the contents will fall out of the packaging. Therefore, you need to adjust the capacity of the MRE.

**Comment by Montimorency** [comment]
One of us could, in a separate PR, restore MRE SUS so that only 2 sugar spawn. But you're right that currently the sugar is unuseable within the MRE (though neither are sporks), because sugar is not edible at the moment. Maybe it would be simpler to expand the carrying volume of mre_package though? An extra 25 ml does the trick.

**Comment by Montimorency** [comment]
BTW, you added drink mixes to MREs, so that technically covers the "sweet water" element. 

I do like the sugar packets for the sake of versimilitude. Looks like 10 ml, no? 
https://www.reddit.com/r/army/comments/v9s4u8/these_curiouslyproportioned_sugar_and_salt/

**Comment by wwkk222208** [comment]
Sure, you can design a container specifically for sugar, but when designing the volume of the outer packaging, it should be noted that the actual volume of the sugar packet will still be affected by the contents, because it should be a "soft" container.

**Comment by wwkk222208** [comment]
I suggest that you adjust the number of sugars in the MRE in this PR, because you need to adjust the outer packaging anyway.

**Comment by Montimorency** [comment]
Take a look at the latest diff. That simple adjustment seems to avoid any problems AFAICT, though more testing on your end is welcome.

**Comment by wwkk222208** [comment]
If the overflow error does not occur again, then there will be no problem. I will make the following adjustments. Thank you for your efforts.

File: data/json/items/comestibles/spice.json [hunk 1]
```diff
@@ -98,9 +98,10 @@
     "price": "5 cent",
     "price_postapoc": "4 cent",
     "material": [ "sugar" ],
-    "volume": "3 ml",
+    "volume": "6 ml",
     "calories": 19,
     "flags": [ "EDIBLE_FROZEN", "NUTRIENT_OVERRIDE" ],
+    "//3": "Restored weight, adjusted volume up to compensate.",
     "//2": "originally was 5g 3ml, and recipes probably use this ratio still;",
     "//": "a tbsp is 4.2g of sugar, meaning the volume needed to be adjusted."
```
Comment (GuardianDll) [inline data/json/items/comestibles/spice.json:106]: should just remove this comments

**Review by GuardianDll — APPROVED** [review]
👍

**Comment by I-am-Erk** [comment]
Unfortunately, changing the volume of sugar causes expected and obnoxious knock-on effects.

```
([slow] ~starting_items)=>   msg := "item sugar could not be put in container jar_glass_sealed when
([slow] ~starting_items)=>   spawning item group sugar_jar_glass_sealed_10_120: not enough space.  This
([slow] ~starting_items)=>   can be resolved either by changing the container or contents to ensure that
([slow] ~starting_items)=>   they fit, or by specifying an overflow behaviour via "on_overflow" on the
([slow] ~starting_items)=>   item 
```

Lemme have a look at what's up here and offer a suggestion.

**Comment by I-am-Erk** [comment]
All right, that looks OK. Probably should consider reducing the amount of sugar that can spawn in a jar so that we don't see a lot of spilled sugar everywhere, but first we should see if we're getting a lot of spilled sugar everywhere or if it's a rare thing.

**Comment by Montimorency** [comment]
According to the itemgroup testing, there shouldn't be any spilling for new spawns. I don't want to discard on existing spawns for evident reasons. If someone can contradict this, we can drop a quick revision.

**Pull request merged**